explorer/ast/bindings.h

```diff
@@ -55,6 +55,11 @@ class Bindings {
   Bindings(BindingMap args, NoWitnessesTag /*unused*/)
       : args_(std::move(args)) {}
 
+  template <typename F>
+  auto Decompose(F f) const {
```

Comment:
Is it possible to be slightly more explicit about the return type?  I'm not sure I agree with these opaque Decompose functions versus calling f(args(), witnesses()) directly, and similar: the lack of comments and function structure (not even std::function?) seems to hide what's going on. Where you have a bunch of opaque Decompose functions for values, why isn't that a switch as we do elsewhere in the explorer?

Note, I'm surprised Decompose isn't mentioned in the PR description given how it differs from the rest of explorer.

Reply:
Where is the Decompose structure explained? I've skimmed over the PR, and I'm still missing it.